theano/gpuarray/opt.py

```diff
@@ -976,11 +981,10 @@ def local_gpua_advanced_incsubtensor(op, context_name, inputs):
 @register_opt('fast_compile')
 @op_lifter([tensor.CAReduce, tensor.Sum, tensor.elemwise.Prod])
 @register_opt2([tensor.CAReduce, tensor.Sum, tensor.elemwise.Prod], 'fast_compile')
-def local_gpua_careduce(op, context_name, inputs):
+def local_gpua_careduce(op, context_name, inputs, outputs):
     if isinstance(op.scalar_op, (scalar.Add, scalar.Mul,
                                       scalar.Maximum, scalar.Minimum)):
 
-        node = op.make_node(*inputs)
```

Comment:
I like that. This seem clean and remove work (so speed up computation). It also prevent giving outputs and clients.

I think we should document in the docstring of the new optimizer class.